### PR TITLE
Refactor prometheus / telemetry functionality

### PR DESCRIPTION
**Pull request by rkachowski**
### Description

Refactors the telemetry implementation to be backed by [Telemetry.Metrics](https://github.com/beam-telemetry/telemetry_metrics). This simplifies the means of defining and collecting metrics, making it easier to monitor the behavior of the application going forwards.

Metrics are now defined in specific modules and passed to a collector process on application startup. 

 ### Other changes

* Updated elixir + otp version in CI tests
    * This was due to some strange formatting difference between the version specified in `.tool-versions` and the CI
    * This caused some dialyzer errors which also needed to be fixed
* Fixed error with tests not loading modules for RPC tests
    * Refactored mapping of module name -> module into `config.exs`
* Ignored block_rewards tests
    * These started failing despite not being changed. Celo protocol doesn't have block rewards - this is now ignored
* Set historian logs to debug on app startup
* Removed logger backend code + error metrics
    * These can be taken from loki directly

     
### Tested

* Passed test suite
* Deployed to rc1staging and saw grafana correctly display dashboard
    * Except for error rate which needs to be changed to use loki

### Issues

 - Relates to https://github.com/celo-org/data-services/issues/577

### CODE REVIEW

File: apps/indexer/lib/indexer/buffered_task.ex [hunk 1]
```diff
@@ -185,6 +185,11 @@ defmodule Indexer.BufferedTask do
     GenServer.call(server, :debug_count)
   end
 
+  @doc false
+  def get_state(server, val) do
```
Comment (carterqw2) [inline apps/indexer/lib/indexer/buffered_task.ex:189]: Where is it used?

File: apps/indexer/lib/indexer/celo/telemetry/instrumentation/indexer.ex [hunk 2]
```diff
@@ -0,0 +1,35 @@
+defmodule Indexer.Celo.Telemetry.Instrumentation do
+  @moduledoc "Metrics for the Indexer application"
+
+  alias Explorer.Celo.Telemetry.Instrumentation
+
+  use Instrumentation
+
+  def metrics do
+    [
+      counter("indexer.import.ingested",
```
Comment (carterqw2) [inline apps/indexer/lib/indexer/celo/telemetry/instrumentation/indexer.ex:10]: Inconsistent with underscores everywhere else.

**Review by carterqw2 — CHANGES_REQUESTED** [review]

File: apps/explorer/lib/explorer/celo/telemetry/helper.ex [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  defp take_import({:transactions, _} = tx), do: tx
+  defp take_import({:blocks, _} = blocks), do: blocks
+  defp take_import({:token_transfers, _} = tt), do: tt
+  defp take_import({:tokens, _} = t), do: t
+  defp take_import({:logs, _} = logs), do: logs
+  defp take_import({:internal_transactions, _} = itx), do: itx
+
+  defp take_import(_), do: nil
+
+  def transform_db_call(_measurements, %{func: function_name}), do: %{function_name => 1}
```
[the remark this answers is not in the document]
Comment (rkachowski) [reply apps/explorer/lib/explorer/celo/telemetry/helper.ex:54]: Yeah it's used to transform the ingestion metrics - https://github.com/celo-org/blockscout/pull/839/files#diff-27ea8b40e2e62180296cfe1086f74f3ccae66b85d6201c189116388376100dc3R152

[the remark this answers is not in the document]
Comment (rkachowski) [reply apps/explorer/lib/explorer/celo/telemetry/helper.ex:54]: Actually that's something else, nice catch

File: apps/indexer/lib/indexer/buffered_task.ex (hunk 1 above)
Comment (rkachowski) [reply apps/indexer/lib/indexer/buffered_task.ex:189]: I started adding new metrics to list common config values of buffered task implementations / fetchers (i.e. emit concurrency + batch size as metric values). I decided not to implement new telemetry events & the functionality to periodically emit them due to the time spent already on this feature. It's something to be added later which shouldn't take too much effort.

File: apps/indexer/lib/indexer/celo/telemetry/instrumentation/indexer.ex (hunk 2 above)
Comment (rkachowski) [reply apps/indexer/lib/indexer/celo/telemetry/instrumentation/indexer.ex:10]: thanks, this was testing the auto definition functionality.

**Review by carterqw2 — APPROVED** [review]
Great job!

**Pull request merged**